src/job/jobHandler.ts

```diff
@@ -635,27 +655,43 @@ export abstract class JobHandler {
     return this._logger;
   }
 
-  protected async previewWebhook(): Promise<AxiosResponse<string>> {
-    const previewWebhookURL = 'https://webhook.gatsbyjs.com/hooks/data_source';
-    const githubUsername = this.currJob.user;
-    const gatsbySiteId = await this._repoEntitlementsRepo.getGatsbySiteIdByGithubUsername(githubUsername);
-    if (!gatsbySiteId) {
-      const message = `User ${githubUsername} does not have a Gatsby Cloud Site ID.`;
-      this._logger.warn('Gatsby Cloud Preview Webhook', message);
-      throw new Error(message);
-    }
+  // Invokes Gatsby Cloud Preview Webhook
+  protected async callGatsbyCloudWebhook(): Promise<void> {
+    const featurePreviewWebhookEnabled = process.env.GATSBY_CLOUD_PREVIEW_WEBHOOK_ENABLED;
+    // Logging for Debugging purposes only will remove once we see the build working in Gatsby.
+    await this.logger.save(
+      this.currJob._id,
+      `${'(GATSBY_CLOUD_PREVIEW_WEBHOOK_ENABLED)'.padEnd(15)}${featurePreviewWebhookEnabled}`
+    );
 
-    const url = `${previewWebhookURL}/${gatsbySiteId}`;
-    return await axios.post(
-      url,
-      {
-        jobId: this.currJob._id,
-        status: this.currJob.status,
-      },
-      {
-        headers: { 'x-gatsby-cloud-data-source': 'gatsby-source-snooty-preview' },
+    try {
+      const previewWebhookURL = 'https://webhook.gatsbyjs.com/hooks/data_source';
+      const githubUsername = this.currJob.user;
+      const gatsbySiteId = await this._repoEntitlementsRepo.getGatsbySiteIdByGithubUsername(githubUsername);
+      if (!gatsbySiteId) {
+        const message = `User ${githubUsername} does not have a Gatsby Cloud Site ID.`;
+        this._logger.warn('Gatsby Cloud Preview Webhook', message);
+        throw new Error(message);
       }
-    );
+
+      const url = `${previewWebhookURL}/${gatsbySiteId}`;
+      const response = await axios.post(
+        url,
+        {
+          jobId: this.currJob._id,
+        },
+        {
+          headers: { 'x-gatsby-cloud-data-source': 'gatsby-source-snooty-preview' },
+        }
+      );
+      await this._jobRepository.updateExecutionTime(this.currJob._id, { gatsbyCloudStartTime: new Date() });
+      await this.logger.save(this.currJob._id, `${'(POST Webhook Status)'.padEnd(15)}${response.status}`);
+    } catch (err) {
+      await this.logger.save(
```

Comment:
minor, should we throw here to kill execution? 

also, would prefer `logger.error` over .save

Reply:
Good call on throwing.

I want to say that `save` in this case will print it out in the build log and in CloudWatch, while `error` would just log it in CloudWatch. It might be good to have this in both places for visibility so that users can surface the webhook error from the log if needed?

Reply:
i'm fine with throwing and saving log 👍 

in theory, this should be a failed job and updated with error status later on